pb/p2pd.proto

```diff
@@ -125,4 +128,42 @@ message ConnManagerRequest {
 
 message DisconnectRequest {
   required bytes peer = 1;
+}
+
+message PSRequest {
+  enum Type {
+    GET_TOPICS = 0;
+    LIST_PEERS = 1;
+    PUBLISH    = 2;
+    SUBSCRIBE  = 3;
+    CANCEL     = 4;
+    NEXT       = 5;
+  }
+
+  required Type type = 1;
+  optional string topic = 2;
+  optional bytes data = 3;
+  optional int64 timeout = 4;
+}
+
+message PSMessage {
+  repeated bytes from = 1;
+  repeated bytes data = 2;
+  repeated bytes seqno = 3;
+  repeated bytes topicIDs = 4;
+  repeated bytes signature = 5;
+  repeated bytes key = 6;
+}
+
+message PSResponse {
+  enum Type {
+    TOPICS = 0;
+    PEERS  = 1;
+    NEXT   = 2;
+  }
+
+  required Type type = 1;
```

Comment:
this is also unnecessary with the messages streamed; the response depends on the request context.

Reply:
We don't want to have the Type for responses to GET_TOPICS or LIST_PEERS?

Reply:
It's implied by the context of the request -- if the request is a GET_TOPICS then the topics field will be set, while if it is a LIST_PEERS then the peers field will be set.

Reply:
I removed the response types